File: hacker-rank/time-conversion.cpp
// https://www.hackerrank.com/challenges/time-conversion/problem
string timeConversion(string s) {
    string hour = s.substr(0, 2);
    if(s[s.length()-2] == 'A' && hour == "12") {
        return "00" + s.substr(2, s.length()-4);
    } else if(s[s.length()-2] == 'P' && hour != "12") {
        int hourInt = std::stoi(hour);
        return std::to_string(hourInt+12) + s.substr(2, s.length()-4);
    }

    return s.substr(0, s.length()-2);
}

File: hacker-rank/time-conversion.java
// https://www.hackerrank.com/challenges/time-conversion/problem
public static String timeConversion(String s) {
    String hour = s.substring(0, 2);

    if(s.charAt(s.length()-2) == 'A' && hour.equals("12")) {
        System.out.println("If 1");
        return "00" + s.substring(2, s.length()-2);
    } else if(s.charAt(s.length()-2) == 'P' && !hour.equals("12")) {
        System.out.println("If 2");
        int hourInt = Integer.parseInt(hour)+12;
        return hourInt + s.substring(2, s.length()-2);
    }

    System.out.println("Else");
    return s.substring(0, s.length()-2);
}
